src/polymer/polymer2-element-scanner.ts

```diff
@@ -65,161 +57,183 @@ export class Polymer2ElementScanner implements JavaScriptScanner {
-  private _document: JavaScriptDocument;
-  // TODO(rictic): write a WarningFeature. Emit them from this scanner.
-  private _warnings: Warning[] = [];
+    const elementFeatures: ScannedPolymerElement[] = [];
+    for (const element of customElements) {
+      elementFeatures.push(this._makeElementFeature(element, document));
+    }
 
-  private _currentElement: ScannedPolymerElement|null = null;
-  private _currentElementNode: estree.Node|null = null;
+    // TODO(rictic): handle normalClasses
 
-  constructor(document: JavaScriptDocument) {
-    this._document = document;
-  }
+    // TODO(rictic): gather up the warnings from these visitors and return them
+    // too.
+    // const warnings = elementDefinitionFinder.warnings;
 
-  enterClassExpression(node: estree.ClassExpression, parent: estree.Node) {
-    if (parent.type !== 'AssignmentExpression' &&
-        parent.type !== 'VariableDeclarator') {
-      return;
-    }
-    const className = astValue.getIdentifierName(
-        parent.type === 'AssignmentExpression' ? parent.left : parent.id);
-    if (className == null) {
-      return;
-    }
-    const element = this._handleClass(node);
-    if (element) {
-      const nodeComments = esutil.getAttachedComment(node) || '';
-      const nodeJsDocs = jsdoc.parseJsdoc(nodeComments);
-      const namespacedClassName =
-          getNamespacedIdentifier(className, nodeJsDocs);
-      element.className = namespacedClassName;
-
-      const summaryTag = jsdoc.getTag(nodeJsDocs, 'summary');
-      element.summary = (summaryTag && summaryTag.description) || '';
-
-      // Set the element on both the namespaced & unnamespaced names so that we
-      // can detect registration by either name.
-      this._possibleElements.set(namespacedClassName, element);
-      this._possibleElements.set(className, element);
-    }
+    return elementFeatures;
   }
 
-  enterClassDeclaration(node: estree.ClassDeclaration) {
-    const element = this._handleClass(node);
-    if (element) {
-      const className = node.id.name;
-      const nodeComments = esutil.getAttachedComment(node) || '';
-      const nodeJsDocs = jsdoc.parseJsdoc(nodeComments);
-      const namespacedClassName =
-          getNamespacedIdentifier(className, nodeJsDocs);
-      element.className = namespacedClassName;
-
-      const summaryTag = jsdoc.getTag(nodeJsDocs, 'summary');
-      element.summary = (summaryTag && summaryTag.description) || '';
-
-      // Set the element on both the namespaced & unnamespaced names so that we
-      // can detect registration by either name.
-      this._possibleElements.set(className, element);
-      this._possibleElements.set(namespacedClassName, element);
+  private _makeElementFeature(
+      element: CustomElementDefinition,
+      document: JavaScriptDocument): ScannedPolymerElement {
+    const node = element.clazz.ast;
+    const docs = element.clazz.doc;
+    const className = element.clazz.namespacedName;
+    let tagName;
```

Comment:
surprised ts isn't getting mad at this, my machine is registering it as an implicit any. Can you define a type?